rotkehlchen/tests/unit/test_general_cache.py

```diff
@@ -36,29 +48,61 @@
         '0xBcca60bB61934080951369a648Fb03DF4F96263C',
         '0x3Ed3B47Dd13EC9a98b44e6204A523E766B225811',
     ],
-    # second 2 are metapool factory pools
-    '0x1F71f05CF491595652378Fe94B7820344A551B8E': [
-        '0x96E61422b6A9bA0e068B6c5ADd4fFaBC6a4aae27',
-        '0x57Ab1ec28D129707052df4dF418D58a2D46d5f51',
-    ],
-    '0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890': [
-        '0xC581b735A1688071A1746c968e0798D642EDE491',
-        '0xD71eCFF9342A5Ced620049e616c5035F1dB98620',
-    ],
 }
 
 CURVE_EXPECTED_GAUGES = {
-    '0x1F71f05CF491595652378Fe94B7820344A551B8E': None,
     '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7': '0xbFcF63294aD7105dEa65aA58F8AE5BE2D9d0952A',
-    '0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890': '0xe8060Ad8971450E624d5289A10017dD30F5dA85F',
     '0xDeBF20617708857ebe4F679508E7b7863a8A8EeE': '0xd662908ADA2Ea1916B3318327A97eB18aD588b5d',
 }
 
+EXPECTED_ADDRESBOOK_ENTRIES_FROM_API = [
+    AddressbookEntry(
+        address=string_to_evm_address('0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7'),
+        name='Curve.fi DAI/USDC/USDT',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xbFcF63294aD7105dEa65aA58F8AE5BE2D9d0952A'),
+        name='Curve gauge for Curve.fi DAI/USDC/USDT',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xDeBF20617708857ebe4F679508E7b7863a8A8EeE'),
+        name='Curve.fi aDAI/aUSDC/aUSDT',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xd662908ADA2Ea1916B3318327A97eB18aD588b5d'),
+        name='Curve gauge for Curve.fi aDAI/aUSDC/aUSDT',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ),
+]
+
+EXPECTED_ADDRESBOOK_ENTRIES_FROM_CHAIN = [
+    AddressbookEntry(
+        address=string_to_evm_address('0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7'),
+        name='3pool curve pool',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xbFcF63294aD7105dEa65aA58F8AE5BE2D9d0952A'),
+        name='Curve gauge for 3pool curve pool',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xDeBF20617708857ebe4F679508E7b7863a8A8EeE'),
+        name='aave curve pool',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        address=string_to_evm_address('0xd662908ADA2Ea1916B3318327A97eB18aD588b5d'),
+        name='Curve gauge for aave curve pool',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ),
+]
+
 
 @pytest.mark.parametrize('have_decoders', [True])
-def test_curve_cache(rotkehlchen_instance):
+@pytest.mark.parametrize('use_curve_api', [True, False])
+def test_curve_cache(rotkehlchen_instance, use_curve_api):
```

Comment:
Okay so why do you do all thse magic mocking functions inside this test and keep an entire json file mock that you hand-crafted instead of just using vcr? We have the system in place now. Keep things simple. Don't waste your time and  write tons of mocking code like this ...

Reply:
Hand-mocking is needed here to limit the number of pools used. By hand-mocking I limit the number of pools to 2. If to mock only with vcr then all curve pools will be queried and it will be difficult to test it.

Reply:
It's not. You can limit the number of pools to 2 with VCR and a patch to the list of pools to query. It's really simple actually.

Reply:
It is easy to patch method queries to the chain this way, but I don't see how this can simplify mocking curve api since curve api endpoints just return a huge list of pools.

Reply:
So you get a list of pools like this: `pools`. And then you want only 2 of those? Can't you do `pools[:2]`?

Reply:
I mean to find a way to write this in a simple way using VCR and minimal mocking. I am not going to implement it for you I am working on something else. My answer is direct to your problem of "returning a huge list of pools while you need 2"